bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BLock.java

```diff
@@ -61,11 +61,11 @@ public synchronized void unlock() {
         }
     }
 
-    private boolean isLockFree() {
+    public boolean isLockFree() {
         return this.current.isEmpty();
     }
 
-    private boolean lockedBySameContext(Strand ctx) {
+    public boolean lockedBySameContext(Strand ctx) {
         return this.current.getLast() == ctx;
```

Comment:
Shouldn't this be a recursive check?

Reply:
Since we only create a new strand when we hit a worker or start, enough to check the current parent.